### PR TITLE
Task: Spring Boot Portfolio Project — Part 10.Add Order model. Executor: Oleksii Bashkirov.

### PR DESCRIPTION
**Pull request by OleksiiBashkirov**
feat: Implement Order and OrderItem entities with soft delete and relationships

- Created Order and OrderItem entities
- Implemented relationships for Order and OrderItem entities
- Added soft delete functionality to Order entity
- Created OrderRepository and OrderItemRepository
- Implemented OrderService and OrderServiceImpl
- Added endpoints for managing orders and order items in OrderController
- Updated Liquibase changelog files:
  - Renamed changelog files for better understanding of their contents
  - Split categories and books_categories tables into separate changelog files

### CODE REVIEW

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                .orElseThrow(() -> new EntityNotFoundException("Order not found by id: "
+                        + orderId));
+        return orderRepository.save(order);
+    }
+
+    @Override
+    public List<OrderItem> getOrderItems(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found by id: "
+                                + orderId));
```
Comment (okuzan) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: repreated logic , extract to a private method

File: src/main/java/mate/academy/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Table(name = "orders")
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE orders SET is_deleted = true WHERE id = ?")
+@SQLRestriction(value = "is_deleted = false")
```
Comment (okuzan) [inline src/main/java/mate/academy/model/Order.java]:
```suggestion
@SQLDelete(sql = "UPDATE orders SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction(value = "is_deleted = FALSE")
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -27,16 +28,16 @@ public class ShoppingCartController {
...
-        var cart = shoppingCartService.getShoppingCartByUserId(userPrincipal.getUser());
+        var cart = shoppingCartService.getShoppingCartByUserId(userPrincipal.getUser().getId());
         return ResponseEntity.ok(cart);
     }
 
     @PostMapping
     public ResponseEntity<Void> addToCart(@AuthenticationPrincipal UserPrincipal userPrincipal,
             @RequestBody AddToCartRequest request) {
-        shoppingCartService.addToCart(userPrincipal.getUser(), request.getBookId(),
+        shoppingCartService.addToCart(userPrincipal.getUser().getId(),request.getBookId(),
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/controller/ShoppingCartController.java]:
```suggestion
        shoppingCartService.addToCart(userPrincipal.getUser().getId(), request.getBookId(),
```

Comment (MentorMateAcademy) [reply src/main/java/mate/academy/controller/ShoppingCartController.java]: formatting code

File: src/main/java/mate/academy/repository/BookRepository.java [hunk 4]
```diff
@@ -11,8 +11,7 @@
 import org.springframework.stereotype.Repository;
 
 @Repository
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/repository/BookRepository.java]:
```suggestion
```

Comment (MentorMateAcademy) [reply src/main/java/mate/academy/repository/BookRepository.java]: this anno is redundant

File: src/main/java/mate/academy/repository/BookRepositoryCustomImpl.java [hunk 5]
```diff
@@ -18,18 +18,23 @@ public class BookRepositoryCustomImpl implements BookRepositoryCustom {
 
     @Override
     public List<Book> findAllBySearchParams(Map<String, List<String>> params) {
-        CriteriaBuilder cb = em.getCriteriaBuilder();
-        CriteriaQuery<Book> cq = cb.createQuery(Book.class);
+        var cb = em.getCriteriaBuilder();
+        var cq = cb.createQuery(Book.class);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/repository/BookRepositoryCustomImpl.java]: short naming is bad practice

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Override
+    public Order updateOrderStatus(Long orderId, Order.Status status) {
+        var order = getOrder(orderId);
+        return orderRepository.save(order);
+    }
+
+    @Override
+    public List<OrderItem> getOrderItems(Long orderId) {
+        var order = getOrder(orderId);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: What profit of using var here?

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -17,46 +16,46 @@
...
 
-    private ShoppingCart createNewShoppingCartForUser(Long userId) {
-        var user = findUserById(userId);
+    private ShoppingCart createShoppingCartForUser(Long userId) {
+        var user = userRepository.findById(userId).orElseThrow(
+                () -> new EntityNotFoundException("User not found by id: " + userId));
         var cart = new ShoppingCart();
         cart.setUser(user);
         return shoppingCartRepository.save(cart);
     }
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: private methods after public

File: src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java [hunk 8]
```diff
@@ -17,46 +16,46 @@
...
-                () -> new EntityNotFoundException("Book not found by id: " + bookId)
-        );
-        var cartItem = getCartItem(quantity, cart, book);
+                () -> new EntityNotFoundException("Book not found by id: " + bookId));
+        var cartItem = createCartItem(quantity, cart, book);
         cart.getCartItems().add(cartItem);
         shoppingCartRepository.save(cart);
     }
 
+    private static CartItem createCartItem(int quantity, ShoppingCart cart, Book book) {
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java]: Why do you need static?

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**